src/mprismetadata.cpp

```diff
@@ -430,7 +439,21 @@ QVariant MprisMetaData::trackId() const
 
 void MprisMetaData::setTrackId(const QVariant &trackId)
 {
-    priv->setMetaData(MetaFieldTrackId, trackId);
+    // Accepts a string, QDBusObjectPath or null.
+    // String must validate as object path or the last element of a path.
+    QVariant value;
+    if (trackId.userType() == qMetaTypeId<QDBusObjectPath>()) {
+        value = trackId.value<QDBusObjectPath>().path();
+    } else if (trackId.isNull()) {
+        // Pass, null QVariant makes D-Bus interface to advertise NoTrack value
+    } else {
+        QString id = trackId.toString();
+        if (!validateTrackIdAsObjectPath(id))
```

Comment:
Is this necessary as a separate function?

Reply:
No, it's not, but I thought it's more readable this way.

Reply:
I know I can squash all the validation on one line of code but then it looks messy. Naming the function should give at least a little bit of idea what is happening there and hopefully the compiler is smart enough to inline it.